source/build.py

```diff
@@ -0,0 +1,169 @@
+import os
+import subprocess
+from os.path import join, getsize
+from HTMLParser import HTMLParser
+
+import pdb, logging
+
+
+# ============================================================================
+
+
+logging.basicConfig(level=logging.DEBUG,
+                    filename='logs.log',
+                    filemode='w')
+
+cwd = str(os.getcwd())
+lsResults = os.listdir(cwd)
+
+isMkdocsInstalled = (
+    subprocess.check_output(['mkdocs', '--version'])[0:15] == 'mkdocs, version')
+
+googleTagManagerContainerId = 'GTM-MNRH4J'
+
+googleTagManagerScriptLines = """
+<!-- Google Tag Manager -->
+  <noscript>
+    <iframe src="//www.googletagmanager.com/ns.html?id={containerId}"
+    height="0" width="0" style="display:none;visibility:hidden"></iframe>
+  </noscript>
+  <script>
+    (function(w,d,s,l,i){{w[l]=w[l]||[];w[l].push({{'gtm.start':
+    new Date().getTime(),event:'gtm.js'}});var f=d.getElementsByTagName(s)[0],
+    j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+    '//www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+    }})(window,document,'script','dataLayer','{containerId}');
+  </script>
+<!-- End Google Tag Manager -->
+""".format(containerId = googleTagManagerContainerId).splitlines()
+
+
+# ============================================================================
+
+
+class PostMkdocsParser(HTMLParser):
+
+    def __init__(self, htmlFileFullPath):
+        self.reset()
+        self.results          = set([])
+        self.targetIsFullHTML = False
+        self.targetFullPath   = htmlFileFullPath
+        self.pruneThisLang    = set([])
+        if '/en/' in self.targetFullPath:
+            self.pruneThisLang.add('ja')
+        if '/ja/' in self.targetFullPath:
+            self.pruneThisLang.add('en')
+
+    def __kill__(self):
+        self.targetIsFullHTML = False
+        self.pruneThisLang.clear()
+        self.results.clear()
+        self.reset()
+
+    def handle_decl(self, data):
+        if data == "DOCTYPE html":
+            self.targetIsFullHTML = True
+
+            if not ('/en/' in self.targetFullPath or '/ja/' in self.targetFullPath):
```

Comment:
This snippet is to mark top-level pages to remove TOC items for both languages.
(eg: /index.html, /search.html)